packages/core/src/node/file-uri.spec.ts

```diff
@@ -64,4 +65,10 @@ describe('file-uri', () => {
         expect(uri.toString(true)).to.be.equal('file:///c:/');
     });
 
+    if (isWindows) {
```

Comment:
First off, having a [file-system test](https://github.com/theia-ide/theia/blob/master/packages/filesystem/src/node/node-filesystem.spec.ts) would be better.

----

If you would like to skip tests, it is better to skip it via `mocha` instead of just omitting it from the test-suite. What do you think?

```ts
    it('from file:///c%3A', function () {
        if (!isWindows) {
            this.skip();
            return;
        }
        const fsPath = FileUri.fsPath('file:///c%3A');
        expect(fsPath).to.be.equal('c:\\');
    });
```

The output will be the following on OS X:
```
  file-uri
    ✓ create -> fsPath -> create should be symmetric
    ✓ fsPath -> create -> fsPath should be symmetric
    ✓ from /
    ✓ from //
    ✓ from c:
    ✓ from /c:
    ✓ from /c:/
    - from file:///c%3A
  7 passing (16ms)
  1 pending
```

The number of (executed) tests will match on the UNIX and Windows CIs.

----

But, since the issue only affects Windows, and you have the `isWindows` guard in the `file-uri` module, it should pass (and run) on every platforms, right? I would just remove the `isWindows` from the test. Thoughts?

And please, add an FS test. If it is not clear what, just let me know, I write it, it should fail, and your patch should fix it ;)

Thanks for your help!

Reply:
If the code processed drive letters always, regardless of the platform running the code, as I originally had it, then indeed we could have just always run the test missing out the os test.  However the test will fail on non-Windows as it will return c: .  We could put c: as the expected result when running on non-Windows but that makes no sense.  I would just leave out the os test in the actual code and always run the test.

I don't understand your comment about FS test.  Is that not what this test is, it fails without the patch, passes with it.

Reply:
Sorry, I replied to your last message before reading up to your earlier messages.  Will do FS test.

Reply:
It is not really possible to put a test in filesystem.spec because those tests execute again the real file system.  The only way to test this patch would be to test against the root of a drive.  However there is no way of obtaining a drive for which we are write access to the root.  Any attempt to make a test run against an actual drive root would cause trouble.

Between the FileUri test and the comments in the code, we are adequately protected against regression.

Reply:
I've modified the test as you suggested with the 'skip'.  We do need to test the os in the test because we are testing it in the code, which means no slash will be added to c: when running on osx or linux.  Note the slightly different form, as there is no  'this' on the callback containing the test code, skip is a method on it.  That does show the '-' when isWindows is false.

Reply:
> We do need to test the os in the test because we are testing it in the code, which means no slash will be added to c: when running on osx or linux.

I am OK with your decision.

> Note the slightly different form, as there is no 'this'

I think you're wrong. I have already [added a snippet example](https://github.com/theia-ide/theia/pull/3419#discussion_r231586664) how to skip a test. You should have the following test case:
```ts
it('from file:///c%3A', function () {
    if (!isWindows) {
        this.skip();
        return;
    }
    const fsPath = FileUri.fsPath('file:///c%3A');
    expect(fsPath).to.be.equal('c:\\');
});
```

Thoughts?

Reply:
Sorry, I had not noticed that you had replaced '() =>' with 'function ()' in order to disable typescript type checking.  I have switched the test to use your form which makes it consistent with the rest of the tests in Theia.

Reply:
Awesome, I am doing a quick review right now...

Reply:
@westbury, no idea what causes the build failure. It passes for me locally. Could you please try to rebase your PR? Thanks!

Reply:
Thanks for looking into the Linux issue.  Unfortunately I don't have a Linux machine but it sounds as though it would not have helped if I had anyway.  I have rebased to current master.